File: 50 Coding Challenge Part I/Python/Challenge20.py
"""
Challenge 20: Rotate an array to the left 1 position
"""


# Function to left-rotate a list by one position
def leftRotateByOne(arr):
    first = arr[0]
    for i in range(len(arr) - 1):
        arr[i] = arr[i + 1]
    arr[-1] = first


# Function to left-rotate a list by `r` positions
def leftRotate(arr, r):
    # base case: invalid input
    if r < 0 or r >= len(arr):
        return

    for i in range(r):
        leftRotateByOne(arr)


# Driver method
myArr = [1, 2, 3, 4, 5]
r = 2
leftRotate(myArr, r)
print(myArr)

File: 50 Coding Challenge Part I/JavaScript/Challenge20.js
// Challenge 20: Rotate an array to the left 1 position

function rotateLeft(arr){
    let first = arr.shift() // removes the first element of the array and returns it
    arr.push(first);

    return arr;
}

// Driver Method
let arr = [1, 2, 3, 4];
console.log(rotateLeft(arr));



